src/test/TestHelper.h

```diff
@@ -1,15 +1,28 @@
 #include "../Iaes_decrypter.h"
 #include "../log.h"
+#include "../common/AdaptiveStream.h"
 #include "../parser/DASHTree.h"
 #include "../parser/HLSTree.h"
 
 
 extern std::string testFile;
 extern std::string effectiveUrl;
+extern std::string lastDownloadUrl;
```

Comment:
Is this common practice to define extern variable? Are they used across classes?

Reply:
I don't know so much about if it's common practice but for this use case these variables are used across TestHelper and TestDASHTree/TestHLSTree. Making them global allows us to avoid having to define what would effectively be the same functions in both TestDASHTree.cpp and TestHLSTree.cpp (`adaptiveTree::download` and `adaptiveStream::download`) and just keeping them in TestHelper

Reply:
Hmmm, it still feels like there should be a C++ way of doing this.

Reply:
No problem, I will investigate :)

Reply:
I've done a lot of research and there's really only 2 ways to go about this from what I gather.

A lot are just recommending to use `extern` to share the variable between source files.

The other method which I've tried and tested is to define a small helper class with static members like so:
```
//TestHelper.h

class staticVars
{
public:
  static std::string testFile;
  static std::string effectiveUrl;
  static std::string lastDownloadUrl;
};
```
```
//TestHelper.cpp

std::string staticVars::testFile;
std::string staticVars::effectiveUrl;
std::string staticVars::lastDownloadUrl;
```

And of course in TestHLSTree.cpp and TestDASHTree.cpp we access by eg. `staticVars::testFile = "foo"`

Which way do you think?

Reply:
Definitely the C++ way using the class!

Reply:
Cool, changes made :)